src/praisonai-sandbox/tests/test_docker_filesystem.py

```
@@ -0,0 +1,98 @@
+    assert sorted(listed) == ["/a.txt", "/sub/b.txt"]
+    for entry in listed:
+        assert ".." not in entry, f"{entry} escapes the sandbox root"
+        assert "private/var" not in entry, f"{entry} leaks the host path"
+
+
+@needs_docker
+def test_the_container_is_still_isolated_from_the_host():
+    """Mounting one directory must not turn into mounting the host."""
+
+    async def check(sandbox):
+        return await sandbox.run_command("ls /Users", shell=True)
```

Ruff (0.16.1)

[error] Function call with `shell=True` parameter identified, security issue

(S604)